addons/arsenal/functions/fnc_isMiscItem.sqf

```diff
@@ -0,0 +1,52 @@
+#include "..\script_component.hpp"
+/*
+ * Author: DartRuffian, LinkIsGrim
+ * Determines if a class is a miscellaneous item or not.
+ *
+ * Arguments:
+ * 0: Item or magazine <STRING>
+ * 1: Item config <CONFIG> (default: nil)
+ * 2: Whether item is a magazine. If true, config should be passed as well <BOOL> (default: false)
+ * 3: If this is being called from preStart. Determines if key will be set even if it's false. If true, config should be passed as well (default: false)
```

Comment:
Why does this arg exist?

Reply:
> 1. So we can potentially skip CBA_fnc_getItemConfig

Just pass `configNull` if you don't have the config. If the the config is `configNull` have `FUNC(isMiscItem)` look it up. I don't like that it's tied to `preStart` for seemingly no reason. 

>     2. So we only cache items that are actually misc items unless it's likely that they will be looked up multiple times. This function will be called for every item in CfgWeapons and CfgMagazines during preStart, which means that the cache is going to be about twice what we consider too fucking big if using USP or something of the sort. By only caching false returns outside of preStart, we get good memory usage while also not getting cache misses on repeat lookups if anything unusual is in a backpack (like a weapon or uniform)

Sounds fair. Should we apply that logic to any other caches too?

Reply:
For me, `preStart` is just too specific, so untying it as you propose is fine.